challenges/02 It's super effective/app.js

```diff
@@ -1 +1,163 @@
+        ice: types[i++],
+        fighting: types[i++],
+        poison: types[i++],
+        ground: types[i++],
+        flying: types[i++],
+        psychic: types[i++],
+        bug: types[i++],
+        rock: types[i++],
+        ghost: types[i++],
+        dragon: types[i++],
+        dark: types[i++],
+        steel: types[i++],
+        fairy: types[i]
+    }
+}
+
+/**
+ * Writes error to stderr in console and then exits
+ * @param  string message Error description
+ */
+function exitWithError(message) {
+    console.error("\x1b[31m", message) //writes error in red
+    process.exit(0) //exit with code 0, because we don't need stacktrace
+}
+
+//Reads attack and defend types, calculates effectiveness of an attack and writes it to console
+rl.question('Please put in attack and defend types like this: typeAttack -> typeDefendA typeDefendB ...\n', (answer) => {
+    const reg = /^\s*(\w+\s+)+\-\>(\s*\w+)+\s*$/i //regular expression to check for right syntax typeA [attackType] -> typeB [typeC] [...]
+    const input = answer.toLowerCase().trim() //trimming our answer, to get rid of spaces before and after first and last letter of answer. 
+        ///Just to ensure everything will be in lower case, we do it extra.
+    rl.close() // disconnect readline from console stdin, because we already have our input data
+    if (input.match(reg) === null) {
+        exitWithError('Your input had wrong syntax')
+    } else {
+        const splittedAttackAndDefend = input.split(/\-\>/)
+        const attackTypesString = splittedAttackAndDefend[0].trim()
+        const defendTypesString = splittedAttackAndDefend[1].trim()
+        let attackType = attackTypesString
+        if (pokemonTypes[attackTypesString] === undefined) {
+            const attackMove = attackTypesString.replace(/\s+/ig, '-') //use replace with regex, to replace all spaces between words with - 
+                ///and use global scope, because otherwise it exits on first word
+            axios.get(`http://pokeapi.co/api/v2/move/${attackMove}`) //get json response from pokemon api by attackMove name
+                .then((response) => {
+                    attackType = response.data.type.name //read effect type from response
+                    checkForPokemonNameAndGetEffectiveness(attackType, defendTypesString)
+                })
+                .catch((error) => {
+                    exitWithError('Could not find attack move or type effect. Error: ' + error.message)
+                })
+        } else {
+            checkForPokemonNameAndGetEffectiveness(attackType, defendTypesString)
+        }
+
+    }
+
+})
+
+/**
+ * Check if we have pokemon name or effect types string and then evaluate effectiveness. If neither of them
+found, then error will be reported
+ * @param  string attackType Attack type
+ * @param  string defendTypesString string of all defend types or pokemon name
+ */
+function checkForPokemonNameAndGetEffectiveness(attackType, defendTypesString) {

```

Comment:
Der Name `checkForPokemonNameAndGetEffectiveness` ist lang, und ein gutes Indiz dafür die Funktion in zwei Funktionen `checkForPokemonName` und `getEffectiveness` aufzuteilen.